torch/nn/parallel/distributed.py

```diff
@@ -697,22 +703,24 @@ def __init__(
         # None after they have been applied by the optimizer. There is no support
         # for setting only some parameter grads to None, this must be done manually
         # by user (and DDP_OVERLAPPED_OPTIM_SET_GRADS_TO_NONE=0 needs to be set.)
-        from torch.distributed.algorithms.ddp_comm_hooks.default_hooks import allreduce_hook
-        from torch.distributed.algorithms.ddp_comm_hooks.optimizer_overlap_hooks import _apply_optim_in_backward_hook
         if any(
             hasattr(p, '_in_backward_optimizers') for p in self.module.parameters()
         ):
             # Remove hooks that apply_optim_in_backward had registered because
             # DDP customizes how optimizer is overlapped with backward due to
             # the allreduce.
             for p in self.module.parameters():
-                for handle in p._optimizer_hook_handles:
+                for handle in getattr(p, '_optimizer_hook_handles', []):
                     handle.remove()
```

Comment:
if we remove the handle, why we require users to pass register_hook=False then? could we enforce users to call apply_optimizer_in_backward() before DDP wrapping? will that resolve the issue without adding the 'register_hook' flag?

Reply:
Discussed this offline, this reasoning makes sense and is the direction we want to converge on, but some internal workloads are running into issues we need to further debug.

For now, we will go with this approach and follow up to ultimately consolidate on either doing `register_hook=False`, or removing the handles.